### PR TITLE
Don't ask for copy_to fields in input

### PR DESCRIPTION
**Pull request by sleitner**

### CODE REVIEW

File: elasticizer/pipelines/extract_postgres.py [hunk 1]
```diff
@@ -50,8 +50,14 @@ class Format(luigi.Task):
...
         with open(self.mapping_file,'r') as fp:
-            mapping_json= json.load(fp,object_pairs_hook=collections.OrderedDict)
-        return mapping_json['properties'].keys()
+            mapping_json = json.load(fp, object_pairs_hook=collections.OrderedDict)
+            fields = mapping_json['properties'].keys()
+            # Automatically remove copy_to fields because they aren't expected from the input
+            copy_targets = set([i.get('copy_to','') for i in mapping_json['properties'].itervalues()])
+            for ct in copy_targets:
+                if ct:
+                    fields.remove(ct)
```
Comment (JeffreyMFarley) [inline elasticizer/pipelines/extract_postgres.py:59]: Is this really generalized elasticizer behavior, or is it merely an artifact of how _we_ are implementing it?

(Hint: I think this belongs in the map generator function in the other repo)

File: elasticizer/pipelines/extract_postgres.py [hunk 2]
```diff
@@ -50,8 +50,14 @@ class Format(luigi.Task):
 
     def _fields_from_mapping(self):
         with open(self.mapping_file,'r') as fp:
-            mapping_json= json.load(fp,object_pairs_hook=collections.OrderedDict)
-        return mapping_json['properties'].keys()
+            mapping_json = json.load(fp, object_pairs_hook=collections.OrderedDict)
+            fields = mapping_json['properties'].keys()
+            # Automatically remove copy_to fields because they aren't expected from the input
+            copy_targets = set([i.get('copy_to','') for i in mapping_json['properties'].itervalues()])
```
Comment (vickumar1981) [inline elasticizer/pipelines/extract_postgres.py]: itervalues() is going away.  i think you want to use just `values()` going forward b/c that's what python 3 will be supporting, and they both return an iterator, i believe.

http://python3porting.com/differences.html

File: elasticizer/pipelines/extract_postgres.py (hunk 1 above)
Comment (sleitner) [inline elasticizer/pipelines/extract_postgres.py:59]: My understanding is that copy_to fields shouldn't exist in the input -- they're created and populated by copy_to calls, so this isn't specific to our use-case.

File: elasticizer/pipelines/extract_postgres.py (hunk 2 above)
Comment (sleitner) [inline elasticizer/pipelines/extract_postgres.py]: :+1:

**Comment by JeffreyMFarley** [comment]
Ok, I see what is happening now.

Next time this code is touched, or a new field added, we should generalize which fields get pulled out of the mapping

**Comment by sleitner** [comment]
Let's discuss -- I think copy_to may be a unique case.

**Pull request merged**